common/src/main/java/com/scottlogic/deg/common/profile/TimeGranularity.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2019 Scott Logic Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.scottlogic.deg.common.profile;
+
+import com.scottlogic.deg.common.ValidationException;
+import com.scottlogic.deg.common.util.Defaults;
+import com.scottlogic.deg.generator.utils.RandomNumberGenerator;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+import java.time.Duration;
+import java.time.LocalTime;
+import java.time.temporal.ChronoUnit;
+import java.util.Objects;
+
+public class TimeGranularity implements Granularity<LocalTime>{
+
+    private final ChronoUnit granularity;
+
+    public TimeGranularity(ChronoUnit granularity) {
+        if (!granularity.isTimeBased()){
+            throw new ValidationException("Granularity for time field must be time based but was " +
+                granularity.toString());
+        }
+        this.granularity = granularity;}
+
+    public static TimeGranularity create(String granularity) {
+        return new TimeGranularity(ChronoUnit.valueOf(granularity.toUpperCase()));
+    }
+
+    @Override
+    public Granularity<LocalTime> getFinestGranularity() {
+        return Defaults.FINEST_TIME_GRANULARITY;
+    }
+
+    @Override
+    public boolean isCorrectScale(LocalTime value) {
+        return trimToGranularity(value).equals(value);
+    }
+
+    @Override
+    public Granularity<LocalTime> merge(Granularity<LocalTime> otherGranularity) {
+        return ((TimeGranularity) otherGranularity).granularity.compareTo(granularity) > 0
+            ? otherGranularity
+            : this;
+    }
+
+    @Override
+    public LocalTime getNext(LocalTime value, int amount) {
+        value = value.truncatedTo(granularity);
+        return value.plus(granularity.getDuration().multipliedBy(amount));
+    }
+
+    @Override
+    public LocalTime getNext(LocalTime value) {
```

Comment:
Nice catch. 

No ```getPrevious()``` had different behaviour when compared with ```getNext()``` in the cases you mentioned. The change to ```getNext``` was required as it was used in an iterator. Whereas ```getPrevious``` was using other validation to ensure that the Time looping would not be an issue.

I have changed get Previous to now return LocalTime.MIN when passed ```getPrevious(LocalTime.MIN)```. This will be useful if we decide to support full sequential mode in a different order.